rita/src/rita_common/tunnel_manager/mod.rs

```diff
@@ -154,941 +218,203 @@ impl Handler<GetPhyIpFromMeshIp> for TunnelManager {
     type Result = ();
-
-    fn handle(&mut self, their_id: OpenTunnel, _: &mut Context<Self>) -> Self::Result {
-        self.open_tunnel(their_id.0, their_id.1).unwrap();
-        ()
+    fn handle(&mut self, peers: PeersToContact, _ctx: &mut Context<Self>) -> Self::Result {
+        trace!("TunnelManager contacting peers");
+        for peer in peers.0.iter() {
+            let res = self.neighbor_inquiry(peer.clone());
+            if res.is_err() {
+                warn!("Neighbor inqury for {:?} failed!", peer);
+            }
+        }
+        // Do not contact manual peers if we are not a gateway
+        if SETTING.get_network().is_gateway {
+            for manual_peer in SETTING.get_network().manual_peers.iter() {
+                let ip = manual_peer.parse::<IpAddr>();
+                let port = SETTING.get_network().rita_hello_port;
+                if ip.is_ok() {
+                    let ip = ip.unwrap();
+                    let socket = SocketAddr::new(ip, port);
+                    let man_peer = Peer {
+                        contact_ip: ip,
+                        ifidx: 0,
+                        contact_socket: socket,
+                    };
+                    let res = self.neighbor_inquiry(man_peer);
+                    if res.is_err() {
+                        warn!("Neighbor inqury for {:?} failed!", manual_peer);
+                    }
+                } else {
+                    let res = self.neighbor_inquiry_hostname(manual_peer.to_string());
+                    if res.is_err() {
+                        warn!("Neighbor inqury for {:?} failed!", manual_peer);
+                    }
+                }
+            }
+        }
     }
 }
 
-pub struct OpenTunnelListener(pub Identity);
-
-impl Message for OpenTunnelListener {
-    type Result = ();
-}
+/// Sets out to contact a neighbor, takes a speculative port (only assigned if the neighbor
+/// responds successfully)
+fn contact_neighbor(peer: Peer, our_port: u16) -> Result<(), Error> {
+    KI.manual_peers_route(
+        &peer.contact_ip,
+        &mut SETTING.get_network_mut().default_route,
+    ).unwrap();
 
-impl Handler<OpenTunnelListener> for TunnelManager {
-    type Result = ();
+    let _res = HTTPClient::from_registry().do_send(Hello {
+        my_id: LocalIdentity {
+            global: SETTING.get_identity(),
+            wg_port: our_port,
+        },
+        to: peer,
+    });
 
-    fn handle(&mut self, their_id: OpenTunnelListener, _: &mut Context<Self>) -> Self::Result {
-        self.open_tunnel_listener(their_id.0).unwrap();
-        ()
-    }
+    Ok(())
 }
 
 impl TunnelManager {
     pub fn new() -> Self {
+        let start = SETTING.get_network().wg_start_port;
+        // TODO we can eliminate the possiblitly of port leaks and save 256kb of ram by
+        // going through and using the operating system to keep track of free ports
+        let mut ports = Vec::<u16>::new();
+        for i in start..65500 {
```

Comment:
Go all the way to `65535`. Also, you can probably just go `let mut ports = (start..65500).collect()`. As mentioned in other parts that touch `free_ports`, the ports most likely should be a set. It would suck big time to iterate this many times to know that a port is not taken.

Reply:
we don't ever have to do that!, we just pop a port off of the list.